lib/Serialization/Serialization.cpp

```diff
@@ -2438,9 +2438,16 @@ void Serializer::writePatternBindingInitializer(PatternBindingDecl *binding,
                                                 unsigned bindingIndex) {
   using namespace decls_block;
   auto abbrCode = DeclTypeAbbrCodes[PatternBindingInitializerLayout::Code];
+
+  StringRef initStr;
+  SmallString<128> scratch;
+  if (binding->hasInitStringRepresentation(bindingIndex)) {
+    initStr = binding->getInitStringRepresentation(bindingIndex, scratch);
```

Comment:
This should probably not be serialized if we don't need it (the same conditions as printing).

I guess in general we need that configuration option that says whether we're making a partial module or a complete module, which would be good enough.